src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const year4 = 'YYYY';
+  const year2 = 'YY';
+
+  const separatedDate = date.split(fromFormat[3]);
+
+  const dateData = {};
+
+  for (let i = 0; i < separatedDate.length; i++) {
+    dateData[fromFormat[i]] = separatedDate[i];
+  }
+
+  if (!dateData[year4]) {
+    if (dateData[year2] < 30) {
+      dateData[year4] = `20${dateData[year2]}`;
+    } else {
+      dateData[year4] = `19${dateData[year2]}`;
+    }
+  } else {
+    dateData[year2] = dateData[year4].slice(2, 4);
+  }
+
+  const result = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    result.push(dateData[toFormat[i]]);
+  }
```

Comment:
This loop does not consider the possibility that 'toFormat' may include 'YY' or 'YYYY'. If 'toFormat' includes 'YY', you should add a condition to slice 'YYYY' to 'YY'. Similarly, if 'toFormat' includes 'YYYY' and the 'fromFormat' was 'YY', you should ensure the conversion has been done correctly before pushing to the result array.